src/Debug.h

```diff
@@ -44,31 +44,21 @@ EXPORT std::ostream &operator << (std::ostream &, const LoweredFunc &);
  * HL_DEBUG_CODEGEN
  */
 
-struct debug {
-    EXPORT static int debug_level;
-    EXPORT static bool initialized;
-    int verbosity;
-
-    debug(int v) : verbosity(v) {
-        if (!initialized) {
-            // Read the debug level from the environment
-            size_t read;
-            std::string lvl = get_env_variable("HL_DEBUG_CODEGEN", read);
-            if (read) {
-                debug_level = atoi(lvl.c_str());
-            } else {
-                debug_level = 0;
-            }
-            initialized = true;
-        }
-    }
+class debug {
+    const bool logging;
+
+public:
+    debug(int verbosity) : logging(verbosity <= debug_level()) {}
 
     template<typename T>
     debug &operator<<(T x) {
```

Comment:
How about T&& and std::forward around the usage while you're editing this code, to prevent copying things when debugging them.